controllers/activemqartemis_reconciler.go

```diff
@@ -1229,6 +1232,33 @@ func (reconciler *ActiveMQArtemisReconcilerImpl) deleteRequestedResource(customR
 	return deleteError
 }
 
+func checkExistingPersistentVolumes(instance *brokerv1beta1.ActiveMQArtemis, client rtclient.Client) {
+	var log = ctrl.Log.WithName("controller_v1beta1activemqartemis")
+
+	var i int32
+	for i = 0; i < instance.Spec.DeploymentPlan.Size; i++ {
+		ordinalString := strconv.Itoa(int(i))
+		pvcKey := types.NamespacedName{Namespace: instance.Namespace, Name: instance.Name + "-" + namer.CrToSS(instance.Name) + "-" + ordinalString}
+		pvc := &corev1.PersistentVolumeClaim{}
+		err := client.Get(context.TODO(), pvcKey, pvc)
+
+		if err == nil {
+			owners := pvc.OwnerReferences
+			if len(owners) > 0 {
+				log.Info("removing owner ref from pvc to avoid potential data loss")
```

Comment:
we need to verify that the owner ref points to the CR, otherwise we may whack an owner reference set by the beta feature at: https://kubernetes.io/blog/2021/12/16/kubernetes-1-23-statefulset-pvc-auto-deletion/

Reply:
checking equality on the GUID will do it: not from here how the owner ref was set in the ss template:
https://github.com/artemiscloud/activemq-artemis-operator/pull/279/files#diff-32e4c985bb650a7028cf649306e140d1cf176b91d4f36dbfbce53083d0f63bd6L2047

Reply:
ok, good point.